docs/lang/articles/basic/difference_between_taichi_and_python_programs.md

```diff
@@ -0,0 +1,77 @@
+---
+sidebar_position: 2
+---
+
+# Difference between taichi and python programs
+
+Although taichi uses Python as the frontend, there are some differences between taichi and python programs.
+Main differences are:
+
+1. Taichi only supports one return statement outside non-static `if`/`for`/`while` scope in the program while Python supports return statements in other places.
+2. Taichi uses lexical scoping (static scoping) while python uses dynamic scoping.
+3. Taichi does not support some of Python's language features.
+
+## Return statement
+
+- If a taichi kernel/function does not have a return statement, it must not have return type annotation.
+- If a taichi kernel has a return statement, it must have return type annotation.
+- If a taichi function has a return statement, return type annotation is optional.
```

Comment:
Let's not suggest this. I believe type annotation for functions will be equally important once we support real function.